File: python/rocketmq/remoting/exception.py
# -*- coding: utf-8 -*-
"""Remoting 异常（对应 org.apache.rocketmq.remoting.exception）。"""
from __future__ import annotations


class RemotingException(Exception):
    pass


class RemotingCommandException(RemotingException):
    pass


class RemotingConnectException(RemotingException):
    def __init__(self, addr: str = ""):
        super().__init__("connect to %s failed" % addr)
        self.addr = addr


class RemotingSendRequestException(RemotingException):
    def __init__(self, addr: str = "", msg: str = ""):
        super().__init__("send request to %s failed: %s" % (addr, msg))
        self.addr = addr


class RemotingTimeoutException(RemotingException):
    def __init__(self, addr: str = "", timeout_millis: int = 0, msg: str = ""):
        super().__init__("wait response on the channel %s timeout, %dms: %s" % (addr, timeout_millis, msg))
        self.addr = addr
        self.timeout_millis = timeout_millis


class RemotingTooMuchRequestException(RemotingException):
    pass


class RemotingNoCodecException(RemotingException):
    pass


class RemotingServerException(RemotingException):
    pass

File: dotnet/src/RocketMQ.Client/Remoting/Exception.cs
// 异常类型（对应 org.apache.rocketmq.remoting.exception.*）。
namespace RocketMQ.Remoting;

/// <summary>基础远程调用异常。</summary>
public class RemotingException : Exception
{
    public RemotingException(string message) : base(message)
    {
    }
}

public class RemotingConnectException : RemotingException
{
    public RemotingConnectException(string message) : base(message)
    {
    }
}

public class RemotingSendRequestException : RemotingException
{
    public RemotingSendRequestException(string message) : base(message)
    {
    }
}

/// <summary>
/// 长轮询超时等「预期内」超时。上层把它当良性 DEBUG 处理——
/// broker 会把下发的 suspend 钳制到自身 brokerSuspendMaxTimeMillis，空闲队列周期性超时是正常行为。
/// </summary>
public class RemotingTimeoutException : RemotingException
{
    public RemotingTimeoutException(string message) : base(message)
    {
    }
}

/// <summary>帧/头部解码失败。</summary>
public class RemotingCommandException : RemotingException
{
    public RemotingCommandException(string message) : base(message)
    {
    }
}

public class RemotingTooMuchRequestException : RemotingException
{
    public RemotingTooMuchRequestException(string message) : base(message)
    {
    }
}
